hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanUploadTask.groovy

```diff
@@ -51,7 +53,28 @@ class AzkabanUploadTask extends DefaultTask {
     if (azkProject.azkabanUrl == null) {
       throw new GradleException("Please set azkaban.url in the .azkabanPlugin.json file in your project directory.");
     }
-    uploadToAzkaban(AzkabanHelper.readSession());
+    authenticateAndUploadToAzkaban(AzkabanHelper.readSession());
+  }
+
+  /**
+   * Authenticates session and uploads the zip file to Azkaban.
+   *
+   * @param sessionId The Azkaban session id. If this is null, an attempt will be made to login to Azkaban.
+   */
+  void authenticateAndUploadToAzkaban(String sessionId) {
+    // If no previous session is available, obtain a session id from server by sending login credentials.
+    if (sessionId == null) {
+      logger.lifecycle("No previous session found. Logging into Azkaban:");
+      logger.lifecycle("Azkaban URL: ${azkProject.azkabanUrl}");
+      logger.lifecycle("Azkaban user name: ${azkProject.azkabanUserName}");
+      logger.lifecycle("Azkaban project name: ${azkProject.azkabanProjName}");
+      logger.lifecycle("Azkaban password: ");
+      sessionId = AzkabanHelper.azkabanLogin(azkProject.azkabanUrl, azkProject.azkabanUserName, System.console().readPassword(" Input>"));
+    }
```

Comment:
Thanks for testing this with a SNAPSHOT build!!! It's possible that depending on the Gradle and JVM version, that the "Input>" will get intermixed with the previous logger.lifecycle messages. However, since you've tested it with a SNAPSHOT build already, let's go with it and see how it works.

In addition, Gradle has made some improvements on how it's logger.lifecycle I/O works in recent versions, so it's possible that it would have been intermixed in previous versions, but is ok now.